azure/scope/machinepoolmachine.go

```diff
@@ -562,6 +591,7 @@ func getNodeByProviderID(ctx context.Context, workloadClient client.Client, prov
 		}
 	}
 
+	// TODO: should this be a not found err if it's not in the list?
```

Comment:
In this case, I think `nil` response with `nil` error is effective in saying something is not in the list. An error might be confused as a 404 from API server.